pkg/load/registryAgent.go

```diff
@@ -61,18 +61,33 @@ func NewRegistryAgent(registryPath string, cycle time.Duration, errorMetrics *pr
 		}
 	}, a.cycle)
 
-	// fsnotify reload
-	registryWatcher, err := fsnotify.NewWatcher()
+	cmMount, err := containsCMMount(a.registryPath)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create new watcher: %v", err)
+		return nil, fmt.Errorf("Failed to walk directory %s: %v", a.registryPath, err)
 	}
-	err = populateWatcher(registryWatcher, a.registryPath)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to populate watcher: %v", err)
+	if cmMount {
```

Comment:
Can we DRY out the code between these two agent init functions? It looks very similar